src/pudl/extract/phmsagas.py

```diff
@@ -0,0 +1,86 @@
+"""Retrieves data from PHMSA natural gas spreadsheets for analysis.
+
+This modules pulls data from PHMSA's published Excel spreadsheets.
+"""
+
+
+from dagster import AssetOut, Output, multi_asset
+
+import pudl.logging_helpers
+from pudl.extract import excel
+
+logger = pudl.logging_helpers.get_logger(__name__)
+
+
+class Extractor(excel.GenericExtractor):
+    """Extractor for the excel dataset PHMSA."""
+
+    def __init__(self, *args, **kwargs):
+        """Initialize the module.
+
+        Args:
+            ds (:class:datastore.Datastore): Initialized datastore.
+        """
+        self.METADATA = excel.Metadata("phmsagas")
+        self.cols_added = []
+        super().__init__(*args, **kwargs)
+
+    def process_final_page(self, df, page):
+        """Drop columns that get mapped to other assets.
+
+        Older years of PHMSA data have one Excel tab in the raw data, while newer data
+        has multiple tabs. To extract data into tables that follow the newer data format
+        without duplicating the older data, we need to split older pages into multiple
+        tables by column. To prevent each table from containing all columns from these
+        older years, filter by the list of columns specified for the page, with a
+        warning.
+        """
+        to_drop = [
+            c
+            for c in df.columns
+            if c not in self._metadata.get_all_columns(page)
+            and c not in self.cols_added
+        ]
+        if to_drop:
+            logger.warning(
+                f"Dropping columns {to_drop} that are not mapped to this asset."
+            )
+            df = df.drop(columns=to_drop, errors="ignore")
+        return df
```

Comment:
This doesn't seem like a bad idea overall but it will result in the extractor needing to read the old excel files for the same chunk of data to extract each of the new parts right?

We could just read the old data as one big squished together table and sort it out during the transform step. Although I do like the idea of not needing to re-org quite as much during the transform step.

Reply:
The parts of the form still exist, it's just that they're all included in one giant Excel sheet instead of being split into Excel sheets in the earlier years. My preference is to form the raw table that we want with the columns that belong to it, with that design being informed by the most recent years of data, and to use the transform step to do cleaning. I'm subclassing this method since I think this is a more strict check than we want for the other data sources, but I think that it is relatively consistent with how we've handled column movements across tabs in other datasets over time.